iceoryx_utils/test/moduletests/test_concurrent_periodic_task.cpp

```diff
@@ -126,49 +166,78 @@ TIMING_TEST_F(PeriodicTask_test, PeriodicTaskWithObjectWithConstructorWithArgume
 
         std::this_thread::sleep_for(SLEEP_TIME);
     }
 
     EXPECT_THAT(PeriodicTaskTestType::callCounter, AllOf(Ge(MIN_RUNS), Le(MAX_RUNS)));
 });
 
-TIMING_TEST_F(PeriodicTask_test, PeriodicTaskWithStdFunction, Repeat(3), [&] {
+TIMING_TEST_F(PeriodicTask_test, PeriodicTaskRunningWithStdFunction, Repeat(3), [&] {
     {
-        using namespace iox::units::duration_literals;
-        concurrent::PeriodicTask<std::function<void()>> sut("Test", 10_ms, PeriodicTaskTestType::increment);
+        concurrent::PeriodicTask<std::function<void()>> sut(INTERVAL, "Test", PeriodicTaskTestType::increment);
 
         std::this_thread::sleep_for(SLEEP_TIME);
     }
 
     EXPECT_THAT(PeriodicTaskTestType::callCounter, AllOf(Ge(MIN_RUNS), Le(MAX_RUNS)));
 });
 
-TIMING_TEST_F(PeriodicTask_test, PeriodicTaskWithMethodCallback, Repeat(3), ([&] {
+// clang-format off
+// due to the `()` enclosing the lambda, clang-format is messing this up
+// the `()` are needed since the `TIMING_TEST_F` macro is messing up the forwarding to the MethodCallback c'tor
+TIMING_TEST_F(PeriodicTask_test, PeriodicTaskRunningWithMethodCallback, Repeat(3), ([&] {
     {
-        using namespace iox::units::duration_literals;
         PeriodicTaskTestType testType;
         concurrent::PeriodicTask<cxx::MethodCallback<void>> sut{
-            "Test", 10_ms, cxx::MethodCallback<void>{testType, &PeriodicTaskTestType::incrementMethod}};
+            INTERVAL, "Test", testType, &PeriodicTaskTestType::incrementMethod};
 
-        std::this_thread::sleep_for(SLEEP_TIME);
+            std::this_thread::sleep_for(SLEEP_TIME);
     }
 
     EXPECT_THAT(PeriodicTaskTestType::callCounter, AllOf(Ge(MIN_RUNS), Le(MAX_RUNS)));
 }));
+// clang-format on
+
+TIMING_TEST_F(PeriodicTask_test, PeriodicTaskWhichIsActiveAppliesNewIntervalAfterStart, Repeat(3), [&] {
+    auto start = std::chrono::steady_clock::now();
+    {
+        constexpr units::Duration WAY_TOO_LARGE_INTERVAL{10 * MAX_RUNS * INTERVAL};
+        concurrent::PeriodicTask<PeriodicTaskTestType> sut(WAY_TOO_LARGE_INTERVAL, "Test");
+
+        sut.start(INTERVAL);
+
+        std::this_thread::sleep_for(SLEEP_TIME);
+    }
+    auto stop = std::chrono::steady_clock::now();
+    auto elapsedTime{std::chrono::duration_cast<std::chrono::milliseconds>(stop - start)};
+
+    EXPECT_THAT(elapsedTime, Le(2 * SLEEP_TIME));
+    EXPECT_THAT(PeriodicTaskTestType::callCounter, AllOf(Ge(MIN_RUNS), Le(MAX_RUNS)));
+});
+
+TIMING_TEST_F(PeriodicTask_test, PeriodicTaskWhichIsExecutingTheCallableIsBlockingOnStop, Repeat(3), [&] {
+    auto start = std::chrono::steady_clock::now();
+    concurrent::PeriodicTask<cxx::function_ref<void()>> sut(INTERVAL, "Test", [] {
+        std::this_thread::sleep_for(SLEEP_TIME);
+    });
+    sut.stop();
+    auto stop = std::chrono::steady_clock::now();
+    auto elapsedTime{std::chrono::duration_cast<std::chrono::milliseconds>(stop - start)};
+
+    EXPECT_THAT(elapsedTime, Ge(SLEEP_TIME));
+});
```

Comment:
What about adding a test case that call `start()` again while the read is still running?

Reply:
that's what `PeriodicTaskWhichIsActiveAppliesNewIntervalAfterStart` is doing

Reply:
Hmm, ok you're calling `start()` indirectly via the other c'tor. That's what I missed. Could we make the user more aware that the behaviour of the two c'tors is **very** different? One does start the other not. I have the feeling folks will do the same mistake as I in the near future (and my future self, too).

Reply:
Any suggestions? An `@attention` doxygen tag?

Reply:
I'm torn inbetween. What about removing the "automatic start c'tor"? An explicit start call might be simpler for most users.

Reply:
Can do that, but then you have the opposite problem to not forget to start it. I'm also not quite sure about this.

Just as food for thought, how about something like this
```c++
class PeriodicTask {
    enum class OnInit {
        STOPPED,
        RUNNING
    }

    PeriodicTask(OnInit onInit, ...)
};

PeriodicTask task{PeriodicTask::OnInit::RUNNING, ...}
```
It could also be the separator between the actual `PeriodicTask` args and the variadic args forwarded to the callable and thus help prevent ambiguities.

Reply:
I like your approach, but it might create too much noise. There can be already quite a lot of args. If we leave out the "automatic start c'tor" won't the user notice this right away and then add start?

Reply:
well, I think it will take as much time as when starting immediately and noticing it's not what you wanted. Maybe even longer because noticing that something is not happening might take longer to realize.

Reply:
added a struct to indicate the automatic or manual start of the task